core/src/main/java/cucumber/runtime/ThreadLocalRunnerSupplier.java

```diff
@@ -13,7 +13,12 @@
     private final GlueSupplier glueSupplier;
     private final EventBus eventBus;
 
-    private final ThreadLocal<Runner> runners = new ThreadLocal<Runner>();
+    private final ThreadLocal<Runner> runners = new ThreadLocal<Runner>() {
+        @Override
+        protected Runner initialValue() {
```

Comment:
This is a cool pattern but was only introduced in 1.8. We're still compiling against  1.7 unfortunately.

Reply:
But it’s in the [Java 7 Docs](https://docs.oracle.com/javase/7/docs/api/java/lang/ThreadLocal.html#initialValue) also could’ve sworn I’d seen it being used elsewhere in the project

Reply:
Ah. You are right!  I had it confused with `ThreadLocal.withInitial`

Reply:
Keep it in!